salt/_states/metalk8s_kubernetes.py

```diff
@@ -159,7 +159,11 @@ def object_present(name, manifest=None, **kwargs):
         name=name_arg, manifest=manifest, old_object=obj, saltenv=__env__, **kwargs
     )
     diff = __utils__["dictdiffer.recursive_diff"](obj, new)
-    ret["changes"] = diff.diffs
+    # Anonymize diff when object has `kind: Secret`
+    if manifest and manifest.get("kind") == "Secret":
```

Comment:
I would use `new` directly instead of `manifest`, normally `new` is always filled